### PR TITLE
Calls snapshotmgr to create snapshot and updates status

### PR DESCRIPTION
**Pull request by xing-yang**
This PR calls snapshot manager to create snapshot and updates snapshot status.

Tested using "velero backup create" and resulted in a successful backup with a local FCD snapshot taken by the snapshot manager.

### CODE REVIEW

File: pkg/backupdriver/backup_driver_controller_base.go [hunk 1]
```diff
@@ -359,7 +369,9 @@ func (rc *backupDriverController) syncSnapshot(key string) error {
 		return err
 	}
 
-	// TODO: Call other function to process create snapshot request
+	err = rc.CreateSnapshot(snapshot)
+	if err != nil {
+	}
```
Comment (lintongj) [inline pkg/backupdriver/backup_driver_controller_base.go:374]: missing `return err`?

**Comment by lintongj** [comment]
Looks good to me. But, there is some pre-existing errors in local test cases. I resolved them in PR #110. So, please hold on the merge of this PR until the PR #110 is merged.

File: pkg/backupdriver/backup_driver_controller_base.go (hunk 1 above)
Comment (xing-yang) [reply pkg/backupdriver/backup_driver_controller_base.go:374]: Oops!  Good catch:).

**Comment by lintongj** [comment]
Hi @xing-yang, looks like you still have conflict in pkg/backupdriver/snapshot_test.go. Please resolve it before merging this PR. BTW, to make sure your change can be built correctly. You can try `make ci` locally.

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/22/

**Comment by xing-yang** [comment]
Hi @lintongj, I submitted my previous update before I learned you've already merged your PR.
Just rebased.  It should pass now:).

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/24/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/25/

**Comment by xing-yang** [comment]
Hi @lintongj, do you know why CI/CD Jenkins failed?  I ran `make ci` locally and it passed.   I checked the CI/CD console logs but it is not clear what is the cause for the failure.  I saw the same failure on the CI/CD run on your PR that was merged yesterday as well: https://github.com/vmware-tanzu/velero-plugin-for-vsphere/pull/110

**Comment by lintongj** [comment]
> Hi @lintongj, do you know why CI/CD Jenkins failed? I ran `make ci` locally and it passed. I checked the CI/CD console logs but it is not clear what is the cause for the failure. I saw the same failure on the CI/CD run on your PR that was merged yesterday as well: #110

@xing-yang I will take a look.

**Comment by lintongj** [comment]
> > Hi @lintongj, do you know why CI/CD Jenkins failed? I ran `make ci` locally and it passed. I checked the CI/CD console logs but it is not clear what is the cause for the failure. I saw the same failure on the CI/CD run on your PR that was merged yesterday as well: #110
> 
> @xing-yang I will take a look.

It is related with backup-driver installation. You don't have to worry about it. Swati's PR #114 will fix it.

File: pkg/backupdriver/backup_driver_controller.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+// CreateSnapshot creates a snapshot of the specified volume, and applies any provided
+// set of tags to the snapshot.
+func (ctrl *backupDriverController) CreateSnapshot(snapshot *backupdriverapi.Snapshot) error {
+	ctrl.logger.Infof("Entering CreateSnapshot: %s/%s", snapshot.Namespace, snapshot.Name)
+
+	objName := snapshot.Spec.TypedLocalObjectReference.Name
+	objKind := snapshot.Spec.TypedLocalObjectReference.Kind
+	if objKind != "PersistentVolumeClaim" {
+		errMsg := fmt.Sprintf("resourceHandle Kind %s is not supported. Only PersistentVolumeClaim Kind is supported", objKind)
+		ctrl.logger.Error(errMsg)
```
[the remark this answers is not in the document]
Comment (xing-yang) [reply pkg/backupdriver/backup_driver_controller.go:36]: but I need to use errMsg on line 37.

**Pull request merged**